staging/src/k8s.io/apiserver/pkg/apis/config/types.go

```diff
@@ -89,4 +89,13 @@ type KMSConfiguration struct {
 	// Timeout for gRPC calls to kms-plugin (ex. 5s). The default is 3 seconds.
 	// +optional
 	Timeout *metav1.Duration
+	// HealthzURL is the url on which kube-apiserver could check the health status of kms-plugin.
+	// +optional
+	HealthzURL string
```

Comment:
Could healthz probes be passed over the plugin `Endpoint` above?

Reply:
I assume that you meant to create a dedicated type for healthz probe. If I understood you correctly, the latest commit accomplishes this.

Reply:
I meant adding a healthz RPC to the KMS plugin itself on the gRPC service.
So you could send those healthz requests over KMSConfiguration.Endpoint (the unix socket) instead of a separate new HTTP endpoint

Reply:
I think this is a good idea, but my survey of the currently implemented plugins shows that they do not have this functionality yet.
This is something we could add later.

Reply:
Do currently implemented plugins already expose a HTTP endpoint for healthz?
If not, you could put in a healthz grpc endpoint and just skip polling health if it returns "unimplemented" (or whatever error grpc will send).

Reply:
Yes, Google CloudKMS Plugin and AWS CloudHSM plugins have implemented healthz. Hence, these plugins could take advantage of this feature right away.

Reply:
Ack, that makes sense then